automation_tools/satellite6/upgrade/tests/existence.py

```diff
@@ -0,0 +1,185 @@
+        return pickle.load(ds)
+
+
+def find_datastore(datastore, component, attribute, search_key=None):
+    """
+    Returns a particular sat component property attribute or all attribute
+    values of component property
+
+    Particular property attribute if search key is provided
+    e.g component='host', search_key='1'(which can be id), attribute='ip'
+    then, the ip of host with id 1 will be returned
+
+    All property attribute values if search key is not provided
+    e.g component='host', attribute='ip'
+    then, List of all the ips of all the hosts will be returned
+
+    :param list datastore: The data fetched from get_datastore function in
+        this module
+    :param str component: The component name of which the property values
+        to find
+    :param str attribute: The property of sat component of which value to be
+        determined
+    :param str search_key: The property value as key of sats given components
+        property
+    :returns str/list: A particular sat component property attribute or list
+        of attribute values of component property
+    """
+    # Lower the keys and attributes
+    component = component.lower() if component is not None else component
+    search_key = search_key.lower() if search_key is not None else search_key
+    attribute = attribute.lower() if attribute is not None else attribute
+    # Fetching Process
+    for i in range(len(datastore)):
+        if component in datastore[i].keys():
+            comp_data = datastore[i][component]
+            break
+    else:
+        raise KeyError(
+            'Unable to find given component \'{0}\' data in satellite.'.format(
+                component))
+    if isinstance(comp_data, list):
+        if (search_key is None) and attribute:
+            attr_values = []
+            for j in range(len(comp_data)):
+                if attribute in comp_data[j].keys():
+                    attr_values.append(comp_data[j][attribute])
+            return attr_values
+        if all([search_key, attribute]):
+            key_index = None
+            for k in range(len(comp_data)):
+                if search_key in comp_data[k].values():
+                    key_index = k
+                    return comp_data[key_index][attribute]
+                    break
+            else:
+                raise KeyError(
+                    'Unable to find search_key \'{0}\' in component \'{1}\' '
+                    'to get \'{2}\' value.'.format(
+                        search_key, component, attribute))
+
+
+# Attributes where 'id' as key to fetch component property data
+attribute_keys = dict.fromkeys(
+    components['non_orged']+['activation-key'], 'id')
+# Attributes with different or specific keys to fetch properties data
+# e.g for content-view there is content view id' and not 'id'
+attribute_keys['content-view'] = 'content view id'
+
+
+# Getting preupgrade and postupgrade data
+predata = get_datastore('preupgrade')
```

Comment:
If you import the module some ssh commands will be run, that is not ideal. The get_datastore should be called inside a task to avoid that.